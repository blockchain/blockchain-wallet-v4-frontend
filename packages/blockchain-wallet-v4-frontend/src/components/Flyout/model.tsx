import React from 'react'
import { FormattedMessage } from 'react-intl'
import moment, { Moment } from 'moment'

import { SBPaymentTypes } from '../../../../blockchain-wallet-v4/src/network/api/simpleBuy/types'
import { RecurringBuyPeriods } from '../../data/components/recurringBuy/types'
import { ActionEnum } from '../../data/types'

const RECURRING_BUY_PERIOD_FETCH = 'RECURRING_BUY_PERIOD_FETCH'

const getPeriodTitleText = (period: RecurringBuyPeriods): React.ReactNode => {
  let text
  switch (period) {
    default:
    case RecurringBuyPeriods.ONE_TIME:
      text = (
        <FormattedMessage
          id='modals.recurringbuys.time_options.one_time'
          defaultMessage='One Time'
        />
      )
      break
    case RecurringBuyPeriods.DAILY:
      text = (
        <FormattedMessage id='modals.recurringbuys.time_options.daily' defaultMessage='Daily' />
      )
      break
    case RecurringBuyPeriods.WEEKLY:
      text = (
        <FormattedMessage id='modals.recurringbuys.time_options.weekly' defaultMessage='Weekly' />
      )
      break
    case RecurringBuyPeriods.BI_WEEKLY:
      text = (
        <FormattedMessage
          id='modals.recurringbuys.time_options.bi_weekly'
          defaultMessage='Twice a Month'
        />
      )
      break
    case RecurringBuyPeriods.MONTHLY:
      text = (
        <FormattedMessage id='modals.recurringbuys.time_options.monthly' defaultMessage='Monthly' />
      )
      break
  }
  return text
}

const getPeriodSubTitleText = (
  period: RecurringBuyPeriods,
  date: string | number = Date.now()
): React.ReactNode => {
  let text
  const startDate = moment(date)
  switch (period) {
    default:
    case RecurringBuyPeriods.DAILY:
      text = <></>
      break
    case RecurringBuyPeriods.WEEKLY:
      text = <>On {startDate.format('dddd')}s</>
      break
    case RecurringBuyPeriods.BI_WEEKLY:
      text = (
        <>
          On the {startDate.format('Do')} and {startDate.add(2, 'weeks').format('Do')}
        </>
      )
      break
    case RecurringBuyPeriods.MONTHLY:
      text = <>On the {startDate.format('Do')}</>
      break
  }
  return text
}

const getPeriodForSuccess = (
  period: RecurringBuyPeriods,
  date: string | number = Date.now()
): React.ReactNode => {
  let text
  const startDate = moment(date)
  switch (period) {
    default:
    case RecurringBuyPeriods.DAILY:
      text = <>every day</>
      break
    case RecurringBuyPeriods.WEEKLY:
      text = <>every {startDate.format('dddd')}</>
      break
    case RecurringBuyPeriods.BI_WEEKLY:
      text = (
        <span style={{ textTransform: 'lowercase' }}>
          {getPeriodSubTitleText(period, date)} of each month
        </span>
      )
      break
    case RecurringBuyPeriods.MONTHLY:
      text = (
        <span style={{ textTransform: 'lowercase' }}>{getPeriodSubTitleText(period, date)}</span>
      )
      break
  }
  return text
}

const getPeriodText = (period: RecurringBuyPeriods): React.ReactNode => {
  let text
  switch (period) {
    default:
    case RecurringBuyPeriods.DAILY:
      text = (
        <FormattedMessage
          id='modals.recurringbuys.timeframe.every_day'
          defaultMessage='Every day'
        />
      )
      break
    case RecurringBuyPeriods.WEEKLY:
      text = <FormattedMessage id='copy.once_a_week' defaultMessage='Once a Week' />
      break
    case RecurringBuyPeriods.BI_WEEKLY:
      text = <FormattedMessage id='copy.twice_a_month' defaultMessage='Twice a Month' />
      break
    case RecurringBuyPeriods.MONTHLY:
      text = <FormattedMessage id='copy.once_a_month' defaultMessage='Once a Month' />
      break
  }
  return text
}

const getActionText = (action: ActionEnum, nextDate: string | number) => {
  let text
  let isToday = false
  let date: Moment | string = moment()

  if (nextDate) {
    isToday = moment(nextDate).calendar().startsWith('Today')
    date = moment(nextDate)
  }

  date = date.format('ddd, MMMM Do')

  switch (action) {
    default:
    case ActionEnum.BUY:
      text = isToday ? (
        <FormattedMessage
          id='scenes.coin.recurringbuy.next_buy_is_today'
          defaultMessage='Next Buy is Today'
        />
      ) : (
        <FormattedMessage
          id='scenes.coin.recurringbuy.next_buy_on_date'
          defaultMessage='Next Buy on {date}'
          values={{ date }}
        />
      )
      break
    case ActionEnum.SELL:
      text = (
        <FormattedMessage
          id='scenes.coin.recurringbuy.next_sell_on_date'
          defaultMessage='Next Sell on {date}'
          values={{ date }}
        />
      )
      break
    case ActionEnum.DEPOSIT:
      text = (
        <FormattedMessage
          id='scenes.coin.recurringbuy.next_deposit_on_date'
          defaultMessage='Next Deposit on {date}'
          values={{ date }}
        />
      )
      break
    case ActionEnum.WITHDRAWAL:
      text = (
        <FormattedMessage
          id='scenes.coin.recurringbuy.next_withdrawal_on_date'
          defaultMessage='Next Withdrawal on {date}'
          values={{ date }}
        />
      )
      break
    case ActionEnum.SWAP:
      text = (
        <FormattedMessage
          id='scenes.coin.recurringbuy.next_swap_on_date'
          defaultMessage='Next Swap on {date}'
          values={{ date }}
        />
      )
      break
  }
  return text
}

const getPaymentMethodText = (paymentMethod: SBPaymentTypes) => {
  let text
  switch (paymentMethod) {
    case SBPaymentTypes.BANK_TRANSFER:
    case SBPaymentTypes.LINK_BANK:
    case SBPaymentTypes.BANK_ACCOUNT:
      text = <FormattedMessage id='copy.bank_account' defaultMessage='Bank Account' />
      break
    case SBPaymentTypes.FUNDS:
      text = <FormattedMessage id='copy.wallet_funds' defaultMessage='Wallet Funds' />
      break
    case SBPaymentTypes.PAYMENT_CARD:
    case SBPaymentTypes.USER_CARD:
      text = <FormattedMessage id='simplebuy.confirm.payment_card' defaultMessage='Credit Card' />
      break
    default:
      text = <>{paymentMethod}</>
      break
  }
  return text
}

export {
  getActionText,
  getPaymentMethodText,
  getPeriodForSuccess,
  getPeriodSubTitleText,
  getPeriodText,
  getPeriodTitleText,
  RECURRING_BUY_PERIOD_FETCH
}
