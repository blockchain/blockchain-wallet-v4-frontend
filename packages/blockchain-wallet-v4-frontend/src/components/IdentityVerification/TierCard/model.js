import { path, prop } from 'ramda'

export const TIERS = {
  1: {
    time: '3',
    level: 'SILVER',
    limit: 'ANNUAL',
    requirements: [
      {
        name: 'EMAIL',
        complete: ({ emailVerified }) => emailVerified === 1
      },
      {
        name: 'NAME',
        complete: ({ userData }) =>
          prop('firstName', userData) && prop('lastName', userData)
      },
      {
        name: 'DOB',
        complete: ({ userData }) => prop('dob', userData)
      },
      {
        name: 'ADDRESS',
        complete: ({ userData }) =>
          path(['address', 'city'], userData) &&
          path(['address', 'country'], userData) &&
          path(['address', 'line1'], userData)
      }
    ]
  },
  2: {
    time: '10',
    level: 'GOLD',
    limit: 'DAILY',
    requirements: [
      {
        name: 'TIER1',
        complete: ({ userTiers }) => userTiers[1].state === 'verified'
      },
      {
        name: 'GOVID',
        complete: ({ userData }) => prop('kycState', userData) !== 'NONE'
      },
      {
        name: 'SELFIE',
        complete: ({ userData }) => prop('kycState', userData) !== 'NONE'
      }
    ]
  }
}
