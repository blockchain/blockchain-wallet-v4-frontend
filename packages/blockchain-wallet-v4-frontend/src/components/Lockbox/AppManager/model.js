export const FAIL_STATUS_TIMEOUT = 10000
export const SUCCESS_STATUS_TIMEOUT = 5000
