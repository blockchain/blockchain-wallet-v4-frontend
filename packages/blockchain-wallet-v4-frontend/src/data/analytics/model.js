export const CUSTOM_DIMENSIONS = {
  CURRENCY_PREFERENCE: 1
}
export const LOCKBOX_EVENTS = {
  INSTALL_APP: ['lockbox', 'apps', 'install'],
  UNINSTALL_APP: ['lockbox', 'apps', 'uninstall'],
  SETTINGS: {
    ADD_DEVICE: ['lockbox', 'settings', 'add_device'],
    AUTHENTICATE_DEVICE: ['lockbox', 'settings', 'check_authenticity'],
    FIRMWARE_UPDATE: ['lockbox', 'settings', 'firmware_update'],
    RENAME_DEVICE: ['lockbox', 'settings', 'rename_device'],
    REMOVE_DEVICE: ['lockbox', 'settings', 'remove_device'],
    SHOW_XPUBS: ['lockbox', 'settings', 'show_xpubs'],
    TAKE_TOUR: ['lockbox', 'settings', 'take_tour']
  }
}
export const TRANSACTION_EVENTS = {
  SEND: ['transactions', 'send'],
  REQUEST: ['transactions', 'request'],
  EDIT_DESCRIPTION: ['transactions', 'edit_description']
}
