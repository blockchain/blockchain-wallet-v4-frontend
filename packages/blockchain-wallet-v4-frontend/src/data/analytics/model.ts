//
// CUSTOM VARIABLES
//
export const CUSTOM_VARIABLES = {
  KYC_TIER: {
    ID: 1,
    NAME: 'kyc_tier'
  },
  CRYPTO_BALANCES: {
    ID: 2,
    NAME: 'crypto_balances'
  }
}

//
// AB TESTS
//
export enum AB_TESTS {}

//
// EVENTS
// format: [event_category, event_action, ?event_name, ?event_value]
//

export const ADS_EVENTS = {
  CLICK_AD: ['navigation', 'click_ad'],
  CLICK_AD_TAX: ['navigation', 'click_ad_tax']
}

export const AB_TEST_EVENTS = {
  DISMISS_VERIFICATION: ['email_ab_test', 'dismiss_verification'],
  EMAIL_VERIFIED: ['email_ab_test', 'email_verified']
}

export const PREFERENCE_EVENTS = {
  GENERAL: {
    ENABLE_BTC_LINKS: 'enable_btc_links'
  },
  SECURITY: {
    VERIFY_DEVICE_ACCEPTED: [
      'preferences',
      'security',
      'verify_device_accepted'
    ],
    VERIFY_DEVICE_EMAIL_SENT: [
      'preferences',
      'security',
      'verify_device_email_sent'
    ],
    VERIFY_DEVICE_REJECTED: [
      'preferences',
      'security',
      'verify_device_rejected'
    ]
  }
}
export const SWAP_EVENTS = {
  ORDER_PREVIEW_ERROR: ['swap', 'order_form', 'order_preview_error'],
  USE_MIN: ['swap', 'order_form', 'use_min'],
  USE_MAX: ['swap', 'order_form', 'use_max']
}
export const TRANSACTION_EVENTS = {
  SEND: ['transactions', 'send'],
  REQUEST: ['transactions', 'request'],
  EDIT_DESCRIPTION: ['transactions', 'edit_description'],
  PAYMENT_REQUEST: ['transactions', 'payment_request'],
  SEND_FAILURE: ['send_failure'],
  VIEW_TX_ON_EXPLORER: ['transactions', 'view_tx_explorer'],
  BITPAY_URL_DEEPLINK: ['transactions', 'bitpay', 'bitpay_url_deeplink'],
  BITPAY_FAILURE: ['transactions', 'bitpay', 'bitpay_payment_failure'],
  BITPAY_SUCCESS: ['transactions', 'bitpay', 'bitpay_payment_success']
}

export const EXCHANGE_EVENTS = {
  BANNER_GET_STARTED: ['exchange', 'homepage', 'homepage_banner_click'],
  BUY_SELL_LINKOUT_CLICKED: ['exchange', 'buy_sell_page', 'linkout_click'],
  BUY_SELL_CONNECT_WALLET_CLICKED: [
    'exchange',
    'buy_sell_page',
    'connect_wallet_click'
  ],
  CONNECT_NOW: ['exchange', 'link_page', 'connect_now_click'],
  LEARN_MORE: ['exchange', 'link_page', 'learn_more_click'],
  LINKED_WALLET_LINKOUT_CLICKED: [
    'exchange',
    'homepage',
    'linked_wallet_linkout_clicked'
  ],
  PROMO: ['exchange', 'send_modal']
}

export const PRICE_CHART_EVENTS = {
  CLICK_BUY_BITCOIN: ['price_chart', 'footer_button', 'buy_bitcoin'],
  CLICK_SWAP_COIN: ['price_chart', 'footer_button', 'swap_coin']
}

export const SB_EVENTS = {
  CLICK_SUPPORT_ARTICLE: ['simple_buy', 'transfer_flyout', 'read_it_now']
}

export const ONE_CLICK_INTEREST = {
  SEEN: ['one_click_interest', 'interest_banner', 'seen'],
  CLICK: ['one_click_interest', 'interest_banner', 'click']
}

export const INTEREST_EVENTS = {
  HOME: {
    CLICK_SUPPORT_ARTICLE: ['interest', 'home', 'learn_more_click'],
    VIEW_TX_SUMMARY: ['interest', 'home', 'view_tx_click']
  },
  SETTINGS: {
    SUPPLY_INFORMATION: ['interest', 'settings', 'supply_information']
  },
  DEPOSIT: {
    SEND_SUCCESS: ['interest', 'deposit', 'send_success'],
    SEND_ONE_CLICK: ['interest', 'deposit', 'send_one_click_success'],
    SEND_FAILURE: ['interest', 'deposit', 'send_failure']
  },
  MODAL: {
    START_EARNING: ['interest', 'deposit modal', 'start_earning'],
    DONT_SHOW_AGAIN: ['interest', 'deposit modal', 'dont show again']
  },
  WITHDRAWAL: {
    REQUEST_SUCCESS: ['interest', 'withdrawal', 'request_success'],
    REQUEST_FAILURE: ['interest', 'withdrawal', 'request_failure']
  }
}
