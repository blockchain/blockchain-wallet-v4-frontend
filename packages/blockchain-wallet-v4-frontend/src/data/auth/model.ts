export const LOGIN_FORM = 'login'
