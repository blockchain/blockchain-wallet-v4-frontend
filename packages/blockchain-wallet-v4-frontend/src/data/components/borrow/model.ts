import {
  CoinType,
  LoanTransactionsStatusType,
  LoanTransactionsType,
  LoanType,
  OfferType
} from 'blockchain-wallet-v4/src/types'

import { convertBaseToStandard } from '../exchange/services'

export const NO_OFFER_EXISTS = 'NO_OFFER_EXISTS'
export const NO_LOAN_EXISTS = 'NO_LOAN_EXISTS'
export const USER_BLOCKED = 'User is from a blocked country or state'

export const getCollateralizationColor = (
  displayName: 'safe' | 'risky' | 'unsafe'
) => {
  switch (displayName) {
    case 'safe':
      return 'green600'
    case 'risky':
      return 'orange600'
    case 'unsafe':
      return 'red600'
  }
}

export const getCollateralAmtRequired = (loan: LoanType, offer: OfferType) => {
  return (
    (offer.terms.collateralRatio - loan.collateralisationRatio) *
    Number(
      convertBaseToStandard(
        loan.principal.amount[0].currency,
        loan.principal.amount[0].amount
      )
    )
  ).toFixed(2)
}

export const getCollateralizationDisplayName = (
  currentCollateral: number,
  offer: OfferType
) => {
  switch (true) {
    case currentCollateral <= offer.callTerms.liquidationHardRatio:
      return 'unsafe'
    case currentCollateral <= offer.callTerms.callTriggerRatio:
      return 'risky'
    default:
      return 'safe'
  }
}

export const fiatDisplayName = (coin: CoinType) => {
  switch (coin) {
    case 'PAX':
      return 'USD'
    default:
      return 'USD'
  }
}

export const isLastTxStatus = (
  statuses: Array<LoanTransactionsStatusType>,
  loan: LoanType,
  loanTransactions: Array<LoanTransactionsType>
): LoanTransactionsType | undefined => {
  let txType

  switch (loan.status) {
    case 'OPEN':
    case 'ON_CALL':
    case 'PENDING_EXECUTION':
    case 'PENDING_COLLATERAL_DEPOSIT':
      txType = 'DEPOSIT_COLLATERAL'
      break
    case 'PENDING_CLOSE':
      txType = 'DEPOSIT_PRINCIPAL_AND_INTEREST'
      break
    default:
      txType = undefined
  }

  if (!txType) return
  const lastDeposit = loanTransactions.find(tx => tx.type === txType)
  return lastDeposit && statuses.indexOf(lastDeposit.status) > -1
    ? lastDeposit
    : undefined
}

export const showBorrowSummary = (loan: LoanType): boolean => {
  switch (loan.status) {
    case 'CLOSED':
    case 'FAILED':
      return false
    default:
      return true
  }
}

export const showCollateralizationStatus = (loan: LoanType): boolean => {
  switch (loan.status) {
    case 'CLOSED':
    case 'FAILED':
    case 'PENDING_EXECUTION':
    case 'PENDING_COLLATERAL_DEPOSIT':
      return false
    default:
      return true
  }
}
