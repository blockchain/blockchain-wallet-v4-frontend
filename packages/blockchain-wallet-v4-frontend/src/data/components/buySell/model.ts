import moment from 'moment'
import { defaultTo, filter } from 'ramda'

import {
  CoinType,
  Limits,
  BSCardType,
  BSOrderActionType,
  BSOrderType,
  BSPairsType,
  BSPaymentTypes,
  SwapOrderType,
  WalletFiatType
} from '@core/types'

import { BankTransferAccountType } from '../brokerage/types'
import { convertBaseToStandard } from '../exchange/services'
import { BSAddCardFormValuesType } from './types'

export const DEFAULT_BS_BALANCE = {
  available: '0',
  pending: '0',
  withdrawable: '0'
}
export const DEFAULT_BS_BALANCES = {}
export const DEFAULT_BS_METHODS = {
  currency: 'EUR' as WalletFiatType,
  methods: []
}

export const POLLING = {
  RETRY_AMOUNT: 30,
  SECONDS: 10
}

export const LIMIT = { max: '10000', min: '500' } as Limits
export const LIMIT_FACTOR = 100 // we get 10000 from API

export const SDD_TIER = 3

export const NO_CHECKOUT_VALS = 'No checkout values'
export const NO_PAIR_SELECTED = 'NO_PAIR_SELECTED'
export const NO_ACCOUNT = 'NO_ACCOUNT'
export const NO_PAYMENT_TYPE = 'NO_PAYMENT_TYPE'
export const NO_FIAT_CURRENCY = 'NO_FIAT_CURRENCY'
export const NO_ORDER_EXISTS = 'NO_ORDER_EXISTS_TO_CONFIRM'

export const BS_CHANGE_EMAIL_FORM = 'sbChangeEmail'
export const BS_CRYPTO_SELECTION = 'sbCryptoSelection'

export const splitPair = (
  pair: BSPairsType
): [WalletFiatType | CoinType, '-', WalletFiatType | CoinType] => {
  return pair.split('-') as [WalletFiatType | CoinType, '-', WalletFiatType | CoinType]
}

export const getOrderType = (order: BSOrderType): BSOrderActionType => {
  return order.side
}

export const getPaymentMethodId = (order: BSOrderType): string | undefined => {
  return order.paymentMethodId
}

export const getBankAccount = (
  order: BSOrderType,
  accounts: BankTransferAccountType[]
): BankTransferAccountType => {
  return filter(
    (b: BankTransferAccountType) => b.state === 'ACTIVE' && b.id === order.paymentMethodId,
    defaultTo([])(accounts)
  )[0]
}

export const getCoinFromPair = (pair: BSPairsType): CoinType => {
  const index = 0
  return splitPair(pair)[index] as CoinType
}

export const getFiatFromPair = (pair: BSPairsType): WalletFiatType => {
  const index = 1
  return splitPair(pair)[index] as WalletFiatType
}

export const getBaseAmount = (order: BSOrderType): string => {
  const orderType = getOrderType(order)

  if (orderType === 'BUY') {
    return convertBaseToStandard(order.outputCurrency as CoinType, order.outputQuantity)
  }
  return convertBaseToStandard(order.inputCurrency as CoinType, order.inputQuantity)
}

export const getCounterAmount = (order: BSOrderType): string => {
  const orderType = getOrderType(order)

  if (orderType === 'BUY') {
    return convertBaseToStandard('FIAT', order.inputQuantity)
  }
  return convertBaseToStandard('FIAT', order.outputQuantity)
}

export const getBaseCurrency = (order: BSOrderType) => {
  const orderType = getOrderType(order)
  return orderType === 'BUY' ? order.outputCurrency : order.inputCurrency
}

export const getCounterCurrency = (order: BSOrderType) => {
  const orderType = getOrderType(order)
  return orderType === 'BUY' ? order.inputCurrency : order.outputCurrency
}
// These methods are being used for just sell p3, since we're release sell first
// Separately from Buy and the order types are different. Once buy is migrated
// to new API, should use methods similar to this
export const getSellBaseAmount = (sellOrder: SwapOrderType): string => {
  const coinCurrency = getCoinFromPair(sellOrder.pair)
  return convertBaseToStandard(coinCurrency as CoinType, sellOrder.priceFunnel.inputMoney)
}

export const getSellCounterAmount = (sellOrder: SwapOrderType): string => {
  return convertBaseToStandard('FIAT', sellOrder.priceFunnel.outputMoney)
}

export const getNextCardExists = (
  existingCards: Array<BSCardType>,
  formValues: BSAddCardFormValuesType
) => {
  return existingCards.find((card) => {
    if (card.state === 'BLOCKED' || card.state === 'FRAUD_REVIEW' || card.state === 'CREATED')
      return false
    if (!card.card) return false
    if (card.card.number !== formValues['card-number'].slice(-4)) return false
    if (
      moment(`${card.card.expireMonth}/${card.card.expireYear}`, 'MM/YYYY').toString() !==
      moment(formValues['expiry-date'], 'MM/YY').toString()
    )
      return false

    return true
  })
}

export const getValidPaymentMethod = (method: BSPaymentTypes) => {
  if (method === BSPaymentTypes.USER_CARD) return BSPaymentTypes.PAYMENT_CARD
  if (method === BSPaymentTypes.BANK_ACCOUNT) return BSPaymentTypes.FUNDS

  return method
}
