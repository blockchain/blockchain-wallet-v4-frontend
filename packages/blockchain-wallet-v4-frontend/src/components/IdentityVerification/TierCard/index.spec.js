import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { assocPath } from 'ramda'

import { TierCard } from './index'
import { INITIAL_TIERS, TIERS_STATES } from 'data/modules/profile/model.js'

const userTiers = INITIAL_TIERS

describe('TierCard', () => {
  it('renders null w/o showBanner', () => {
    const component = shallow(<TierCard tier={1} userTiers={userTiers} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with column view', () => {
    const component = shallow(
      <TierCard tier={2} userTiers={userTiers} column />
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('should render continue button if tier has started', () => {
    const verifyIdentity = jest.fn()
    const component = shallow(
      <TierCard
        tier={1}
        userTiers={userTiers}
        userData={{ tiers: { selected: 1 } }}
        verifyIdentity={verifyIdentity}
      />
    )
    const button = component.find('TierCard__ActionButton')
    expect(button).toHaveLength(1)
    expect(button.children().prop('defaultMessage')).toBe('Continue')
    button.simulate('click')
    expect(verifyIdentity).toHaveBeenCalledTimes(1)
  })

  it('should render get started button if tier has not started', () => {
    const verifyIdentity = jest.fn()
    const component = shallow(
      <TierCard
        tier={1}
        userTiers={userTiers}
        userData={{ tiers: { selected: 0 } }}
        verifyIdentity={verifyIdentity}
      />
    )
    const button = component.find('TierCard__ActionButton')
    expect(button).toHaveLength(1)
    expect(button.children().prop('defaultMessage')).toBe('Unlock Silver')
    button.simulate('click')
    expect(verifyIdentity).toHaveBeenCalledTimes(1)
  })

  it('should render swap now button if tier is verified', () => {
    const tiers = assocPath([1, 'state'], TIERS_STATES.VERIFIED, userTiers)
    const goToSwap = jest.fn()
    const component = shallow(
      <TierCard tier={1} userTiers={tiers} goToSwap={goToSwap} />
    )
    const button = component.find('TierCard__ActionButton')
    expect(button).toHaveLength(1)
    expect(button.children().prop('defaultMessage')).toBe('Swap Now')
    button.simulate('click')
    expect(goToSwap).toHaveBeenCalledTimes(1)
  })

  it('should no button if tier state is neither none nor verified', () => {
    const tiers = assocPath([1, 'state'], TIERS_STATES.PENDING, userTiers)
    const component = shallow(<TierCard tier={1} userTiers={tiers} />)
    expect(component.find('TierCard__ActionButton')).toHaveLength(0)
  })
})
