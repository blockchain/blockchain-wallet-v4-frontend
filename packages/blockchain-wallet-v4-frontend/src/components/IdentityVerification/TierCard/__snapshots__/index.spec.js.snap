// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TierCard renders correctly with column view 1`] = `
<TierCard__Wrapper
  className=" column"
>
  <TierCard__Announcement
    color="white"
    size="18px"
    uppercase={true}
    weight={500}
  >
    <FormattedMessage
      defaultMessage="Get Free Crypto"
      id="components.identityverification.tiercard.getfreecrypto"
      values={Object {}}
    />
  </TierCard__Announcement>
  <TierCard__Container>
    <TierCard__Header
      color="marketing-primary"
      uppercase={true}
    >
      <FormattedMessage
        defaultMessage="Gold Level"
        id="components.identityverification.tiercard.gold"
        values={Object {}}
      />
    </TierCard__Header>
    <TierCard__Content>
      <TierCard__Row>
        <TierCard__Column>
          <Text
            altFont={false}
            capitalize={false}
            color="marketing-secondary"
            cursor="inherit"
            italic={false}
            opacity={1}
            size="32px"
            uppercase={false}
            weight={400}
          >
            $25,000
          </Text>
          <Text
            altFont={false}
            capitalize={false}
            color="textBlack"
            cursor="inherit"
            italic={false}
            opacity={1}
            size="14px"
            style={
              Object {
                "marginTop": "8px",
              }
            }
            uppercase={false}
            weight={400}
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="14px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Daily Swap Limit"
                id="components.identityverification.tiercard.dailylimit"
                values={Object {}}
              />
            </Text>
          </Text>
          <Text
            altFont={false}
            capitalize={false}
            color="gray-3"
            cursor="inherit"
            italic={false}
            opacity={1}
            size="14px"
            style={
              Object {
                "marginTop": "7px",
              }
            }
            uppercase={false}
            weight={400}
          >
            <FormattedMessage
              defaultMessage="Takes {time} min"
              id="components.identityverification.tiercard.timing"
              values={
                Object {
                  "time": "10",
                }
              }
            />
          </Text>
        </TierCard__Column>
        <TierCard__Column>
          <TextGroup
            inline={true}
            key="0"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-3"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Silver Level +"
                id="components.identityverification.tiercard.silverplus"
                values={Object {}}
              />
            </Text>
          </TextGroup>
          <TextGroup
            inline={true}
            key="1"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Gov't Issued ID"
                id="components.identityverification.tiercard.id"
                values={Object {}}
              />
            </Text>
            <Icon
              color="success"
              cursor={false}
              name="check"
              size="12px"
              style={
                Object {
                  "marginLeft": "5px",
                }
              }
              weight={400}
            />
          </TextGroup>
          <TextGroup
            inline={true}
            key="2"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              style={
                Object {
                  "display": "inline-block",
                  "marginBottom": "26px",
                }
              }
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Portrait Photo"
                id="components.identityverification.tiercard.portraitphoto"
                values={Object {}}
              />
            </Text>
            <Icon
              color="success"
              cursor={false}
              name="check"
              size="12px"
              style={
                Object {
                  "marginLeft": "5px",
                }
              }
              weight={400}
            />
          </TextGroup>
        </TierCard__Column>
      </TierCard__Row>
    </TierCard__Content>
    <TierCard__ActionButton
      fullwidth={true}
      jumbo={true}
      nature="primary"
    >
      <FormattedMessage
        defaultMessage="Unlock Gold"
        id="components.identityverification.tiercard.goldcta"
        values={Object {}}
      />
    </TierCard__ActionButton>
  </TierCard__Container>
</TierCard__Wrapper>
`;

exports[`TierCard renders null w/o showBanner 1`] = `
<TierCard__Wrapper
  className=""
>
  <TierCard__Container>
    <TierCard__Header
      color="marketing-primary"
      uppercase={true}
    >
      <FormattedMessage
        defaultMessage="Silver Level"
        id="components.identityverification.tiercard.silver"
        values={Object {}}
      />
    </TierCard__Header>
    <TierCard__Content>
      <TierCard__Row>
        <TierCard__Column>
          <Text
            altFont={false}
            capitalize={false}
            color="marketing-secondary"
            cursor="inherit"
            italic={false}
            opacity={1}
            size="32px"
            uppercase={false}
            weight={400}
          >
            $1,000
          </Text>
          <Text
            altFont={false}
            capitalize={false}
            color="textBlack"
            cursor="inherit"
            italic={false}
            opacity={1}
            size="14px"
            style={
              Object {
                "marginTop": "8px",
              }
            }
            uppercase={false}
            weight={400}
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="14px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Annual Swap Limit"
                id="components.identityverification.tiercard.annuallimit"
                values={Object {}}
              />
            </Text>
          </Text>
          <Text
            altFont={false}
            capitalize={false}
            color="gray-3"
            cursor="inherit"
            italic={false}
            opacity={1}
            size="14px"
            style={
              Object {
                "marginTop": "7px",
              }
            }
            uppercase={false}
            weight={400}
          >
            <FormattedMessage
              defaultMessage="Takes {time} min"
              id="components.identityverification.tiercard.timing"
              values={
                Object {
                  "time": "3",
                }
              }
            />
          </Text>
        </TierCard__Column>
        <TierCard__Column>
          <TextGroup
            inline={true}
            key="0"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Verified Email"
                id="components.identityverification.tiercard.email"
                values={Object {}}
              />
            </Text>
          </TextGroup>
          <TextGroup
            inline={true}
            key="1"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Name"
                id="components.identityverification.tiercard.name"
                values={Object {}}
              />
            </Text>
          </TextGroup>
          <TextGroup
            inline={true}
            key="2"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Date of Birth"
                id="components.identityverification.tiercard.dob"
                values={Object {}}
              />
            </Text>
          </TextGroup>
          <TextGroup
            inline={true}
            key="3"
            nowrap={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
          >
            <Text
              altFont={false}
              capitalize={false}
              color="gray-5"
              cursor="inherit"
              italic={false}
              opacity={1}
              size="12px"
              uppercase={false}
              weight={400}
            >
              <FormattedMessage
                defaultMessage="Address"
                id="components.identityverification.tiercard.address"
                values={Object {}}
              />
            </Text>
          </TextGroup>
        </TierCard__Column>
      </TierCard__Row>
    </TierCard__Content>
    <TierCard__ActionButton
      fullwidth={true}
      jumbo={true}
      nature="primary"
    >
      <FormattedMessage
        defaultMessage="Unlock Silver"
        id="components.identityverification.tiercard.silvercta"
        values={Object {}}
      />
    </TierCard__ActionButton>
  </TierCard__Container>
</TierCard__Wrapper>
`;
